src/_guides/libraries/library-tour.md

```diff
@@ -1035,6 +1035,38 @@ For more information, see
 [Exceptions](/guides/language/language-tour#exceptions)
 (in the language tour) and the [Exception API reference.][Exception]
 
+### Weak references and finalizers
+Dart is a [garbage-collected][] language,
+which means that any Dart object
+that isn't referenced by at least one other object
+will eventually be disposed by the garbage collector. 
+This default behavior may not be desireable in
+some scenarios involving native resources or 
+if the target object cannot be sub-classed.
+
+A [WeakReference][]
+stores a reference to the target object
+without increasing the reference count
+considered by the garbage collector.
+Another option is to use an [Expando][]
+to add properties to an object
+without affecting how they are garbage-collected.
+
+A [Finalizer][] can be used to execute a callback function
+after an object is no longer referenced.
+It is not guaranteed to execute this callback, however.
+
+A [NativeFinalizer][]
+provides stronger guarantees
+for interacting with native code using [dart:ffi][].
+Its callback will be called at least once
+after the object is no longer referenced.
+It can be used to close native resources
+such as a database connection or open files.
+
+To ensure that an object won't be finalized
+or garbage collected too early,
+classes can implement the [Finalizable][] interface.
```

Comment:
I'm not sure if it is the right level of detail for this entry, but as a reader I'm a little confused on what "too early" means. Maybe we should expand more explicitly on what guarantee implementing `Finalizable` provides? Something like being alive the entire scope of the variable (or whatever it guarantees).

Reply:
That's a good point, I added another sentence to clarify.